setup.py

```diff
@@ -26,6 +26,7 @@
         "ldap3>=2.5.2,<2.6.0",
         "anymarkup>=0.7.0,<0.8.0",
         "python-gitlab>=1.11.0,<1.12.0",
+        "GitPython>=2.1.14,<3.0.0",
```

Comment:
I'm not sure we need this, you are using git in two ways: 1) to git clone and 2) to get a file from a previous commit. These two things can be easily done with subprocess, no?

for reference:
```
git show <commit>:<path>
```